crates/flipperzero/examples/view_dispatcher.rs

```diff
@@ -0,0 +1,130 @@
+//! Demonstrates use of the ViewDispatcher module.
+//!
+//! This app prompts the user for a name then says hello.
+
+#![no_main]
+#![no_std]
+
+extern crate alloc;
+extern crate flipperzero_alloc;
+extern crate flipperzero_rt;
+
+use alloc::boxed::Box;
+use core::ffi::{c_char, c_void, CStr};
+use flipperzero::furi::string::FuriString;
+use flipperzero_rt::{entry, manifest};
+use flipperzero_sys as sys;
+
+const RECORD_GUI: *const c_char = sys::c_string!("gui");
+
+manifest!(name = "Rust ViewDispatcher example");
+entry!(main);
+
+enum AppView {
+    Widget = 0,
+    TextInput = 1,
+}
+
+struct App {
+    name: [i8; 16],
```

Comment:
You might want to use [`core::ffi::c_char`](https://doc.rust-lang.org/stable/core/ffi/type.c_char.html) here if this is a C-string.

Reply:
This is a buffer provided to the `TextInput` to populate with a null-terminated C-string. I'm using a Rust array so that Rust allocates it as part of the Box and takes care of the lifetime.

Reply:
Ah. `core::ffi::c_char` should be the same as `i8` (usually). So `[c_char; 16]` should behave exactly the same.

It just helps make it clear that this array is being used as an array of `char` rather than an array of numbers.

Reply:
Ok, I understand now. I'll change the buffer type to `[c_char; 16]`.